### ISSUE TITLE
Reactivepower HMS1800 6,553.5 var

### ISSUE THREAD

**Issue by arittel** [tbnobody/OpenDTU#987]
### What happened?

The Reactivepower at the HMS-1800 shows the wrong value.
It stay's at 6,553.5 var there for also the Powerfactor gets shown als 0.002

![image](https://github.com/tbnobody/OpenDTU/assets/123459855/f6514ab1-0467-49b2-b126-3bca047d49c0)

The Value also get's send over the MQQT to ioBroker.

### To Reproduce Bug

Just connect to a HMS-1800.

### Expected Behavior

Powerfactor should about 1 and the ReactivePower should be around 0.0 - 0.2 var

### Install Method

Self-Compiled

### What git-hash/version of OpenDTU?

23.5.31

**Comment by tbnobody** [comment]
Seems to be the same as https://github.com/tbnobody/OpenDTU/wiki/FAQ#q-with-hms-2000-the-reactive-power-shows-unrealistic-value-65535 and therefor not fixable.

**Comment by tbnobody** [comment]
Going to close this issue as there is no chance to fix it as it is related to the firmware on the inverter itself.

**Comment by broth-itk** [comment]
![Screenshot at 2023-08-15 11-22-50](https://github.com/tbnobody/OpenDTU/assets/2356596/f7f3cdeb-b108-4e4b-8d2f-4fb3d6175380)

According to Hoymiles, this bug should have been solved among many other issues.
I upgraded to the latest version 1.0.20 but the values are now changing between 

6.553,5 var, 0,0 var and 6.553,4 var

Sometimes a useful value of 0.2 is shown.

Question: Are we having an issue with signed/unsigned integers here?

**Comment by stefan123t** [comment]
@broth-itk how did you upgrade the firmware from 1.0.14 to 1.0.20 ?
Do you have an Hoymiles DTU too, which model ?

Would you or someone else on the old 1.0.14 firmware still be able to trace the firmware update performed by Hoymiles
or did you receive the firmware images to update them yourself ?
We are very interested in firmware images and detailled traces of the update procedure / protocol :wink:

**Comment by broth-itk** [comment]
@stefan123t The Upgrade was done with DTU-Pro-S after Hoymiles Support has initiated the upgrade process.

I would have been very happy to help and capture the communication, but somehow I remembered that the project is only meant to collect data ;-)

Anyway, with the original DTU we can still capture a trace when a new firmware version is released.
For now we can capture the method to set a grid profile or any other communication.

My OpenDTU has to be set to "listen only" and dump all to serial.

According to Hoymiles, the overflow is present in 1.0.14 for HMS-2000-4T (or maybe others).
1.0.14 on my HMS-350-1T is behaving fine as far as I can tell.

Other issues noticed with 1.0.14 on HMS-2000-4T were: MPPT inconsistencies or a very long waiting time (20 minutes and more) until providing power to the grid. I will need to check my grafana diagrams to confirm that those were solved.

In order not to mix up topics, maybe stick to the signed/unsigned 16 bit integer issue.
